src/log/Entry.h

```diff
@@ -40,6 +47,7 @@ class Entry {
   time m_stamp;
   pthread_t m_thread;
   short m_prio, m_subsys;
+  unsigned int m_global_thread_id; // for Log::dump_recent()
```

Comment:
I feel like it would be much simpler to:

- save the thread's name in thread_local storage (once)
- copy the 128 (16 byte) value in this Entry

Reply:
It's in the `Entry` itself. It can be copied out into the `recent_pthread_ids` set.

Reply:
I want to add that since there is strong interest to get this fix into older releases, I would strongly recommend this simple solution. I suspect the performance and memory cost to be inconsequential. It can be explored afterwards whether optimizations are worthwhile.

Reply:
I had indeed started this PR with copying the thread name into the `Entry` object ... but was discouraged by Radek to do so due to memory and time consumption in the hot path

Reply:
Nope, it wasn't just _copying_ of 16 bytes; it was a call to the lock-protected `Thread::fetch_thread_name()` (https://github.com/ceph/ceph/pull/60058#discussion_r1783630273) which spurs the concern.

IIUC Patrick is arguing for the simpler approach where the thread name is copied from TLS by the `Entry` constructor:

> Maybe instead of this complex, lock-protected container every `Thread` could cache its name in a `thread_local` variable from which the `Entry` constructor would simply copy?

Reply:
Okay, then we wouldn't require any thread local storage at all.
We could simply call `ceph_pthread_getname(pthread_self(), ...)` in the `Entry` constructor and save it for future reference.
Does this sound reasonable ?

Reply:
This is what I understand:

- Save the thread name (once) in TLS during thread creation (one call to pthread_getname_np(...))
- In the ctor for Entry, fetch the thread name from TLS (no locking required)
- copy the thread name in the Entry object 

So, repeated calls to pthread_getname_np(...) is avoided and the overhead is the 16byte copy.

Reply:
Discussed this with @mchangir a bit and it's boiling down to deciding if we want to store the thread name in TLS vs calling pthread_getname_np() when creating Entry object. As @rzarzynski mentioned, the pthread call might have a more optimized implementation. So, I'm really on the fence hence to choose between the two.